forage-android/src/main/java/com/joinforage/forage/android/collect/BTPinCollector.kt

```diff
@@ -64,21 +96,23 @@ internal class BTPinCollector(
                 return ForageApiResponse.Failure(
                     listOf(
                         ForageError(
-                            // Error code hardcoded as 400 because of lack of information
-                            400,
+                            httpStatusCode,
                             error.code,
                             error.message
                         )
                     )
                 )
-            } catch (e: JSONException) { }
+            } catch (_: JSONException) { }
             logger.i(
                 "[BT] Received successful response from BasisTheory",
                 attributes = mapOf(
                     "merchant_ref" to merchantAccount,
                     "payment_method_ref" to paymentMethodRef
                 )
             )
```

Comment:
this logic's a bit confusing for me. So if I understand, we're:
- checking if the response was successful
  - if it was successful, try getting the error in a try block, which throws an error if not available
  - catch the thrown error from when error isn't available, and use that to determine a successful response
  
If that's correct, could it be more intuitive to do it the same way we do in the js repo?
```
   val forageResponse = response.getOrNull()
            try {
                // This will return null if an error isn't available, instead of throwing
                val forageApiError = ForageApiError.ForageApiErrorMapper.from(forageResponse.toString())
                
                if (forageApiError) { // handleError}
                // Else handle success?
                }
```

Reply:
I'm going to make a new ticket to attempt to address this because it is a little weird as of now.